tool/src/js/mod.rs

```diff
@@ -75,24 +113,43 @@ pub fn gen_bindings(env: &Env, outs: &mut HashMap<String, String>) -> fmt::Resul
 /// a struct, at the boundary as well.
 fn used_types<'env>(
     custom_type: &'env ast::CustomType,
+    is_declaration: bool,
```

Comment:
Question: Can you explain why this function needs to have different behavior based on whether or not we are in a decl file? Consider adding this to the docs string above.

Reply:
In decl files, we want struct fields (if the type is a struct), parameter types, and return types. In js files, we also want struct fields (if struct), but we don't care about importing parameter types. However, js does care about any nested enums, which get converted at the boundary. This got cleaned up a lot in my next pr and has a nice doc string.